src/features/island/hud/components/buildings/Buildings.tsx

```diff
@@ -214,8 +219,17 @@ export const Buildings: React.FC<Props> = ({ onClose }) => {
                 key={name}
                 onClick={() => setSelectedName(name)}
                 image={
-                  ITEM_ICONS(state.island.type, state.season.season)[name] ??
-                  ITEM_DETAILS[name].image
+                  ITEM_ICONS(
+                    state.island.type,
+                    state.season.season,
+                    name in BUILDING_UPGRADES
+                      ? state[
+                          makeUpgradableBuildingKey(
+                            name as UpgradableBuildingType,
+                          )
+                        ].level
+                      : undefined,
+                  )[name] ?? ITEM_DETAILS[name].image
```

Comment:
Maybe this would be more readable if it was broken up a bit

```ts
const { island, season } = state;
const isUpgradable = name in BUILDING_UPGRADES;
const upgradableKey = isUpgradable ? makeUpgradableBuildingKey(name as UpgradableBuildingType) : undefined;
const buildingLevel = upgradableKey 
  ? state[upgradableKey].level 
  : undefined;

const itemIcon = ITEM_ICONS(island.type, season.season, buildingLevel)[name] 
  ?? ITEM_DETAILS[name].image;
```